core/src/test/java/org/infinispan/persistence/sifs/SoftIndexFileStoreRestartTest.java

```diff
@@ -44,28 +45,37 @@ SoftIndexFileStoreRestartTest fileSize(int fileSize) {
       return this;
    }
 
+   SoftIndexFileStoreRestartTest hasPassivation(boolean passivation) {
+      this.hasPassivation = passivation;
+      return this;
+   }
+
    @Override
    public Object[] factory() {
       return Stream.of(CacheMode.DIST_SYNC, CacheMode.LOCAL)
-            .flatMap(type ->
-                  Stream.builder()
-                        .add(new SoftIndexFileStoreRestartTest().fileSize(1_000).cacheMode(type))
-                        .add(new SoftIndexFileStoreRestartTest().fileSize(10_000).cacheMode(type))
-                        .add(new SoftIndexFileStoreRestartTest().fileSize(320_000).cacheMode(type))
-                        .add(new SoftIndexFileStoreRestartTest().fileSize(2_000_000).cacheMode(type))
-                        .add(new SoftIndexFileStoreRestartTest().fileSize(DataConfiguration.MAX_FILE_SIZE.getDefaultValue()).cacheMode(type))
-                        .build()
+            .flatMap(type -> Stream.of(Boolean.TRUE, Boolean.FALSE)
+                  .flatMap(hasPassivation -> Stream.of((SoftIndexFileStoreRestartTest) new SoftIndexFileStoreRestartTest().hasPassivation(hasPassivation).fileSize(1_000).cacheMode(type))
+                        .flatMap(sifsrt ->
+                           Stream.builder()
+                                 .add(sifsrt.fileSize(1_000))
+                                 .add(sifsrt.fileSize(10_000))
+                                 .add(sifsrt.fileSize(320_000))
+                                 .add(sifsrt.fileSize(2_000_000))
+                                 .add(sifsrt.fileSize(DataConfiguration.MAX_FILE_SIZE.getDefaultValue()))
```

Comment:
I guess `fileSize` should return a new instance? All tests are running with the same size.

Reply:
Ah, yes good catch.

Reply:
I didn't complete my changes here as the original new was even wrong, heh